ReportingServicesTools/Functions/Admin/Set-RsDatabase.ps1

```diff
@@ -123,11 +153,37 @@ function Set-RsDatabase
             $password = $DatabaseCredential.GetNetworkCredential().Password
         }
         #endregion Validating authentication and normalizing credentials
-        
+
+        #region Validating admin authentication and normalizing credentials
+        $adminUsername = ''
+        $adminPassword = $null
+
+        # default is Windows
+        $isSQLAdminAccount = ($AdminDatabaseCredentialType -like "SQL")
+
+        # we do not allow service account - only Windows and SQL
+        if ($AdminDatabaseCredentialType -like 'serviceaccount')
+        {
+            throw "Can only use Admin Database Credentials Type of 'Windows' or 'SQL'"
+        }
+
+        # must have credentials passed
+        if ($AdminDatabaseCredentialType -like 'sql')
```

Comment:
just use the variable you created on 162 $isSQLAdminAccount

Reply:
Updated.